src/mono/wasm/runtime/startup.ts

```diff
@@ -842,48 +880,62 @@ let configLoaded = false;
         const configResponse = await runtimeHelpers.fetch_like(resolveSrc);
         const configData: MonoConfig = (await configResponse.json()) || {};
+        if (configData.environment_variables && typeof (configData.environment_variables) !== "object")
+            throw new Error("Expected config.environment_variables to be unset or a dictionary-style object");
+
         // merge
         configData.assets = [...(config.assets || []), ...(configData.assets || [])];
+        configData.environment_variables = { ...(config.environment_variables || {}), ...(configData.environment_variables || {}) };
         config = runtimeHelpers.config = Module.config = Object.assign(Module.config as any, configData);
-
-        // normalize
-        config.environment_variables = config.environment_variables || {};
-        config.assets = config.assets || [];
-        config.runtime_options = config.runtime_options || [];
-        config.globalization_mode = config.globalization_mode || "auto";
-        if (config.enable_debugging)
-            config.debug_level = config.enable_debugging;
-
-        if (typeof (config.environment_variables) !== "object")
-            throw new Error("Expected config.environment_variables to be unset or a dictionary-style object");
+        normalize();
 
         if (Module.onConfigLoaded) {
             try {
                 await Module.onConfigLoaded(<MonoConfig>runtimeHelpers.config);
+                normalize();
             }
             catch (err: any) {
                 _print_error("MONO_WASM: onConfigLoaded() failed", err);
                 throw err;
             }
         }
-        runtimeHelpers.diagnostic_tracing = !!runtimeHelpers.config.diagnostic_tracing;
-        runtimeHelpers.enable_debugging = runtimeHelpers.config.enable_debugging ? runtimeHelpers.config.enable_debugging : 0;
-        configLoaded = true;
+        afterConfigLoaded.promise_control.resolve();
     } catch (err) {
         const errMessage = `Failed to load config file ${configFilePath} ${err}`;
         abort_startup(errMessage, true);
         config = runtimeHelpers.config = Module.config = <any>{ message: errMessage, error: err, isError: true };
         throw err;
     }
+
+    function normalize() {
+        // normalize
+        config.environment_variables = config.environment_variables || {};
+        config.assets = config.assets || [];
+        config.runtime_options = config.runtime_options || [];
+        config.globalization_mode = config.globalization_mode || "auto";
+        if (config.enable_debugging)
+            config.debug_level = config.enable_debugging;
+        if (config.debug_level === undefined && BuildConfiguration === "Debug") {
+            config.debug_level = -1;
+        }
+        runtimeHelpers.diagnostic_tracing = !!runtimeHelpers.config.diagnostic_tracing;
+        runtimeHelpers.enable_debugging = runtimeHelpers.config.enable_debugging ? runtimeHelpers.config.enable_debugging : 0;
```

Comment:
And this also seems disconnected from the debug_level/enable_debugging above?

Reply:
I have open different PR on which I'm removing `enable_debugging` completely.

Reply:
It gone by now